### PR TITLE
Upgrading engine UI

### PR DESCRIPTION
**Pull request by armandojaga**
This PR is a huge improvement on UI consistency and overall appearance of the engine

### CODE REVIEW

File: Gameplay/misc/CameraPosChange.h [hunk 1]
```diff
@@ -22,15 +22,16 @@ namespace Hachiko
...
+			GameObject* _player{};
+			PlayerCamera* _camera{};
 			bool _is_inside = false;
 			SERIALIZE_FIELD(GameObject*, _objective);
 			SERIALIZE_FIELD(bool, _do_look_ahead);
 			SERIALIZE_FIELD(float, _speed);
 			SERIALIZE_FIELD(float, _duration);
 			SERIALIZE_FIELD(float3, _relative_position);
 			SERIALIZE_FIELD(float3, _rotation);
+			SERIALIZE_FIELD(float4, _rotationf4);
```
Comment (baransrc) [inline Gameplay/misc/CameraPosChange.h:34]: I believe this was forgotten as it's never used. Let's delete this.

File: Source/src/components/ComponentMeshRenderer.cpp [hunk 2]
```diff
@@ -18,13 +18,12 @@
 
 #include <debugdraw.h>
 
-Hachiko::ComponentMeshRenderer::ComponentMeshRenderer(GameObject* container) 
-    : Component(Type::MESH_RENDERER, container)
+Hachiko::ComponentMeshRenderer::ComponentMeshRenderer(GameObject* container) :
+    Component(Type::MESH_RENDERER, container)
```
Comment (baransrc) [inline Source/src/components/ComponentMeshRenderer.cpp:22]: To be honest, I find this initializer list style inferior to the previous one, as the previous looks like the following when there are many members:

```
Namespace::Class::Class() 
    : Base()
    , some_member(2)
    , another_member(0.0f)
    , yet_another_member(false) {}
    
```

Is it possible that we define this in our clang coding style?

File: Source/vendors/MathGeoLib/Math/Quat.cpp [hunk 3]
```diff
@@ -537,14 +537,14 @@ void SetQuatFrom(Quat &q, const M &m)
...
-	assume(!m.HasNegativeScale());
+	// assume(m.IsColOrthogonal());
+	// assume(m.HasUnitaryScale());
+	// assume(!m.HasNegativeScale());
 	SetQuatFrom(*this, m);
 
 #ifdef MATH_ASSERT_CORRECTNESS
 	// Test that the conversion float3x3->Quat->float3x3 is correct.
-	mathassert(this->ToFloat3x3().Equals(m, 0.01f));
+	//mathassert(this->ToFloat3x3().Equals(m, 0.01f));
```
Comment (baransrc) [inline Source/vendors/MathGeoLib/Math/Quat.cpp:547]: We can revert these changes as in a prior commit we already disabled these asserts centrally from MathGeoLibConfig.h.

**Review by baransrc — CHANGES_REQUESTED** [review]
Thanks for the good work!! After 2 small changes, we are good to go. Also is it possible for you to write us a little document to fit the new coming ui to the new framework that you have created? I see there are some labeling conventions and new widgets that need to be used instead of vanilla ImGui ones.

File: Source/src/core/Scene.cpp [hunk 4]
```diff
@@ -339,22 +339,26 @@ void Hachiko::Scene::FogConfig::SaveFogParams(YAML::Node& node)
...
     ImGuiUtils::CompactOpaqueColorPicker("Fog Color", fog.color.ptr());
-    ImGui::DragFloat("Global Density", &fog.global_density, 0.001, 0.f, 1.f);
-    ImGui::DragFloat("Height Falloff", &fog.height_falloff, 0.001, 0.f, 1.f);
-    ImGui::PopItemWidth();
+    Widgets::DragFloatConfig cfg;
+    cfg.speed = 0.001f;
+    cfg.min = 0.0f;
+    cfg.max = 1.0f;
+    DragFloat("Global Density", fog.global_density, &cfg);
+    DragFloat("Height Falloff", fog.height_falloff, &cfg);
```
Comment (monikamnc) [inline Source/src/core/Scene.cpp:361]: I think in this script changes we lost some decimals on Ambient Light and Fog.
Not sure if you did because it's not so crucial and were so tiny modifications that we can't percieve.
If other reviewers didn't notice or don't mind about them we can leave them like this.
I was just curious about if we needed them or not.

**Comment by monikamnc** [comment]
Also maybe take a look into the sonarcloud comments to see if we can decrease some codesmells or the duplicated lines it is complaining about (?)

File: Gameplay/misc/CameraPosChange.h (hunk 1 above)
Comment (armandojaga) [reply Gameplay/misc/CameraPosChange.h:34]: oh yeah, that was test code, thanks for the catch

File: Source/vendors/MathGeoLib/Math/Quat.cpp (hunk 3 above)
Comment (armandojaga) [reply Source/vendors/MathGeoLib/Math/Quat.cpp:547]: this is also code that was merely for testing, thanks for the review

File: Source/src/components/ComponentMeshRenderer.cpp (hunk 2 above)
Comment (armandojaga) [reply Source/src/components/ComponentMeshRenderer.cpp:22]: the format that it has is defined, I just ran the formatter on some files, we can change that directly on the config file, but that is outside the scope of this

File: Source/src/core/Scene.cpp (hunk 4 above)
Comment (armandojaga) [reply Source/src/core/Scene.cpp:361]: Do you see anything weird? It's supposed to behave the same as before, same limits and same increment

**Review by baransrc — APPROVED** [review]

File: Source/src/core/Scene.cpp (hunk 4 above)
Comment (monikamnc) [reply Source/src/core/Scene.cpp:361]: It's only that we had 0.000 to tweak and now it shows 0.00
On that two specific ones the decimals make a big change I think, but as you consider.

Comment (armandojaga) [reply Source/src/core/Scene.cpp:362]: thank you, I updated the format, so that the UI reflects the correct precision

**Comment by armandojaga** [comment]
> Also maybe take a look into the sonarcloud comments to see if we can decrease some codesmells or the duplicated lines it is complaining about (?)

This is going to be tricky, maybe I can address this in a separate PR

**Review by monikamnc — APPROVED** [review]
Thank you for your work!
